OmniKit/PumpManager/PodComms.swift

```diff
@@ -41,52 +44,168 @@ class PodComms: CustomDebugStringConvertible {
+                    switch podCommsError {
+                    // These errors can happen some times when the responses are not seen for a long
+                    // enough time. Automatically retrying using the already incremented packet # can
+                    // clear this condition without requiring any user interaction for a pairing failure.
+                    case .podAckedInsteadOfReturningResponse, .noResponse:
+                        if didRetry == false {
+                            didRetry = true
+                            log.debug("sendPairMessage to retry using updated packet #%d", transport.packetNumber)
+                            continue // the transport packet # is already advanced for the retry
+                        }
+                    default:
+                        break
+                    }
+                }
+                throw error
+            }
+
+            if let fault = response.fault {
+                log.error("Pod Fault: %{public}@", String(describing: fault))
+                if let podState = self.podState, podState.fault == nil {
+                    self.podState!.fault = fault
+                }
+                throw PodCommsError.podFault(fault: fault)
+            }
+
+            guard let config = response.messageBlocks[0] as? VersionResponse else {
+                log.error("sendPairMessage unexpected response: %{public}@", String(describing: response))
+                let responseType = response.messageBlocks[0].blockType
+                throw PodCommsError.unexpectedResponse(response: responseType)
+            }
+
+            guard config.address == address else {
+                log.error("sendPairMessage unexpected address return of %{public}@ instead of expected %{public}@",
+                  String(format: "04X", config.address), String(format: "%04X", address))
+                throw PodCommsError.invalidAddress(address: config.address, expectedAddress: address)
+            }
+
+            // If we previously had podState, verify that we are still dealing with the same pod
+            if let podState = self.podState, (podState.lot != config.lot || podState.tid != config.tid) {
+                // Have a new pod, could be a pod change w/o deactivation (or we're picking up some other pairing pod!)
+                log.error("Received pod response for [lot %u tid %u], expected [lot %u tid %u]", config.lot, config.tid, podState.lot, podState.tid)
+                throw PodCommsError.podChange
+            }
+
+            // Checking RSSI
+            let maxRssiAllowed = 59         // maximum RSSI limit allowed
+            let minRssiAllowed = 30         // minimum RSSI limit allowed
+            if let rssi = config.rssi, let gain = config.gain {
+                let rssiStr = String(format: "Receiver Low Gain: %d.\nReceived Signal Strength Indicator: %d", gain, rssi)
+                log.default("%s", rssiStr)
```

Comment:
This snippet should be kept here for any users who might want to customize a different pairing RSSI enforcement policy and for any developers working on RileyLinks and similar devices. We would also need to add back a variable declaration for rssiTesting (or perhaps some alternate name) that can be edited by the user to enable.

                if rssiTesting {
                    throw PodCommsError.debugFault(str: rssiStr)
                }

Reply:
This is easy enough for someone to add back in. I don't want the code to end up littered with testing conditionals that readers of the code have to bushwack past to try to understand the code.